### PR TITLE
depth-estimator block

### PR DESCRIPTION
**Pull request by meet-rocking**
Adding Depth-estimator block in core:

- input: file block (image path)
- output: list (saved image path).
- model: hf-model, will download the model during the run time.

### CODE REVIEW

File: frontend/core/blocks/depth-estimator/specs.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "information": {
+    "id": "depth-v2",
+    "name": "Depth Anything",
+    "description": "Performs depth estimation on an input image using the Depth-Anything-V2-Small model.\n\nThis function uses the Hugging Face Transformers pipeline to load the \n'depth-anything/Depth-Anything-V2-Small-hf' model for depth estimation. It processes \nthe input image and generates a depth map, which is then saved as 'depth.jpg'.\n\nInputs:\n    img_path (str): Path to the input image file.(use file core block)\n\nOutputs:\n    dict: A dictionary containing the following key-value pair:\n        'depth' (list): A list containing the filename of the saved depth map ('depth.jpg').\n\nRequirements:\n    - transformers library\n    - PIL (Python Imaging Library)\n    - 'depth-anything/Depth-Anything-V2-Small-hf' model from Hugging Face\n\nExample:\n    result = compute('path/to/your/image.jpg')\n    print(result)  # Output: {'depth': ['depth.jpg']}\n\nNote:\n    The depth map is saved in the current working directory as 'depth.jpg'.\n    Ensure you have sufficient disk space and write permissions.",
```
Comment (amansingh0330) [inline frontend/core/blocks/depth-estimator/specs.json:5]: is this is the correct way to have the description

File: frontend/core/blocks/depth-estimator/specs.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  "outputs": {
+    "depth": {
+      "type": "Any",
+      "connections": []
+    }
+  },
+  "action": {
+    "container": {
+      "image": "depth-v2",
+      "version": "depth-estimator-latest",
```
Comment (amansingh0330) [inline frontend/core/blocks/depth-estimator/specs.json:28]: it should just be latest not depth-estimator-latest

**Review by amansingh0330 — COMMENTED** [review]
The package-lock.json can be removed 

Apart from that there are couple of minor changes required

File: frontend/core/blocks/depth-estimator/specs.json (hunk 2 above)
Comment (meet-rocking) [reply frontend/core/blocks/depth-estimator/specs.json]: I have make change as per your suggestion, thanks

File: frontend/core/blocks/depth-estimator/specs.json (hunk 1 above)
Comment (meet-rocking) [reply frontend/core/blocks/depth-estimator/specs.json]: I have the change the description, thanks.

**Review by amansingh0330 — APPROVED** [review]

**Pull request merged**